### PR TITLE
RavenDB-16626 - forward fixed AcknowledgeSubscriptionBatchWhenDBisBeingDeletedShouldThrow and backported it

### PR DESCRIPTION
**Pull request by lastav5**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-16626

### Additional description

The new exception `SubscriptionMessageTypeException` inherits from `RavenException` which leads to a different code path in `ShouldTryToReconnect`  than it did when it was a simple `Exception`. The change caused the test to fail almost constantly in v5.3.

### Type of change

- Bug fix

### How risky is the change?

- Low

### CODE REVIEW

**Review by karmeli87 — APPROVED** [review]

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs [hunk 1]
```diff
@@ -341,7 +341,7 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
                 if (connectionStatus.Type == SubscriptionConnectionServerMessage.MessageType.Error)
                     message += $". Exception: {connectionStatus.Exception}";
 
-                throw new SubscriptionMessageTypeException(message);
+                throw new Exception(message);
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:344]: Nope

Comment (ppekrol) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:344]: This will be handled after the release, reverts recent change

Comment (garayx) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:344]: why not add `SubscriptionMessageTypeException` to `ShouldTryToReconnect` swtich?

Comment (lastav5) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:344]: Either that or let it inherit straight from Exception

Comment (arekpalinski) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:344]: I think that for consistency it should inherit from `SubscriptionException` as other subscription specific exceptions. It needs to be correctly handled though.

**Pull request merged**